### PR TITLE
feat: add user language

### PR DESCRIPTION
**Pull request by capJavert**
- [x] add user language column
- [x] add support for language in `newUser` endpoint
- [x] update boot to support language
- [x] users schema updates to support reading/writing to language

https://github.com/dailydotdev/heimdall/pull/98

### CODE REVIEW

File: __tests__/private.ts [hunk 1]
```diff
@@ -547,6 +568,48 @@ describe('POST /p/newUser', () => {
...
+        name: usersFixture[0].name,
+        image: usersFixture[0].image,
+        username: usersFixture[0].username,
+        email: usersFixture[0].email,
+        experienceLevel: 'foo',
+        language: 'klingon',
+      })
+      .expect(200);
+
+    expect(body).toEqual({ status: 'failed', reason: 'MISSING_FIELDS' });
```
Comment (rebelchris) [inline __tests__/private.ts:611]: We actually need to check maybe with Viktor what happens now if we throw some error to heimdall?
Not really sure it persists back to the end user. (which would mean probably half broken user exists somewhere)

File: src/entity/user/User.ts [hunk 2]
```diff
@@ -239,15 +243,23 @@ export type AddUserData = Pick<
...
 
+const checkLanguage = (language?: string): boolean => {
+  return !!language && validLanguages.includes(language as ContentLanguage);
+};
+
 const checkRequiredFields = (data: AddUserData): boolean => {
-  if (data?.username && !data?.experienceLevel) {
+  if (
+    data?.username &&
+    (!data?.experienceLevel || !checkLanguage(data?.language))
```
Comment (rebelchris) [inline src/entity/user/User.ts:261]: Was just wondering if we should maybe split them into 2 internal ifs?

File: src/migration/1724168427191-UserLanguage.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class UserLanguage1724168427191 implements MigrationInterface {
+  name = 'UserLanguage1724168427191';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "user" ADD "language" text`,
```
Comment (rebelchris) [inline src/migration/1724168427191-UserLanguage.ts:8]: We don't want to set a default?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Minor feedback, think most is fine, but blocking for validation.
Also please sync with Viktor on deployment of heimdall and once done we need a e2e auth check: https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/458096761/Critical+flows

File: src/migration/1724168427191-UserLanguage.ts (hunk 3 above)
Comment (capJavert) [reply src/migration/1724168427191-UserLanguage.ts:8]: I was thinking we might want to differentiate users that selected language and those legacy ones that did not yet? So we can (for example) run a campaign to update language at later date?

File: __tests__/private.ts (hunk 1 above)
Comment (capJavert) [reply __tests__/private.ts:611]: It should be the same as when we introduced experienceLevel, so should be no problem?

File: src/migration/1724168427191-UserLanguage.ts (hunk 3 above)
Comment (capJavert) [reply src/migration/1724168427191-UserLanguage.ts:8]: Either way in API users will fallback to english when nothing is set

File: __tests__/private.ts (hunk 1 above)
Comment (capJavert) [reply __tests__/private.ts:611]: On the other thought, maybe experienceLevel was introduced as required later due to extension, I can just make it optional for now until adoption/experiment is over?

Comment (rebelchris) [reply __tests__/private.ts:611]: It needs to be optional for flows where we subscribe first -> then ask for it.
I think email is that exception if I remember correct.
 (Actually not sure how we fixed that now for expLevel?) CC: @nensidosari

Comment (capJavert) [reply __tests__/private.ts:611]: Required experienceLevel was introduced here https://github.com/dailydotdev/daily-api/pull/2034, see thread for more info

Comment (capJavert) [reply __tests__/private.ts:611]: I think we don't need language to be required until experiment concludes either way?

Comment (rebelchris) [reply __tests__/private.ts:611]: Oh I see: `if (data?.username` fixes the email users :) 
Hmm yeah just have it optional I guess for now.

File: src/migration/1724168427191-UserLanguage.ts (hunk 3 above)
Comment (rebelchris) [reply src/migration/1724168427191-UserLanguage.ts:8]: Ah yeah that's good :)

File: src/entity/user/User.ts (hunk 2 above)
Comment (capJavert) [reply src/entity/user/User.ts]: Split it now

File: __tests__/private.ts (hunk 1 above)
Comment (capJavert) [reply __tests__/private.ts:591]: Made optional here 58bc9a9ab43f7f34d3074256a418e27825be06b4 but still validates it if provided

**Review by rebelchris — APPROVED** [review]
Looks good!

File: __tests__/private.ts [hunk 4]
```diff
@@ -547,6 +548,48 @@ describe('POST /p/newUser', () => {
...
+      .set('Content-type', 'application/json')
+      .set('authorization', `Service ${process.env.ACCESS_SECRET}`)
+      .send({
+        id: usersFixture[0].id,
+        name: usersFixture[0].name,
+        image: usersFixture[0].image,
+        username: usersFixture[0].username,
+        email: usersFixture[0].email,
+        experienceLevel: 'foo',
+        language: 'klingon',
```
Comment (nensidosari) [inline __tests__/private.ts:587]: 😆

**Review by nensidosari — APPROVED** [review]

File: __tests__/private.ts (hunk 4 above)
Comment (capJavert) [reply __tests__/private.ts:587]: until we introduce it then we will change the test 😆

**Pull request merged**